src/cephadm/cephadm

```diff
@@ -946,16 +946,21 @@ def port_in_use(ctx, port_num):
     # type: (CephadmContext, int) -> bool
     """Detect whether a port is in use on the local machine - IPv4 and IPv6"""
     logger.info('Verifying port %d ...' % port_num)
-    try:
-        s = socket.socket(socket.AF_INET, socket.SOCK_STREAM)
-        attempt_bind(ctx, s, '0.0.0.0', port_num)
-
-        s = socket.socket(socket.AF_INET6, socket.SOCK_STREAM)
-        attempt_bind(ctx, s, '::', port_num)
-    except OSError:
-        return True
-    else:
-        return False
+    def _port_in_use(af: socket.AddressFamily, address: Literal['0.0.0.0', '::']) -> bool:
+        try:
+            s = socket.socket(af, socket.SOCK_STREAM)
+            attempt_bind(ctx, s, address, port_num)
+        except OSError as e:
+            # If IPv4 or IPv6 is not supported, do not automatically return "occupied"
+            if e.errno == errno.EAFNOSUPPORT:
+                return False
+            return True
```

Comment:
what about

```suggestion
            if e.errno == errno.EADDRINUSE:
                return True
            return False
```

Reply:
Thought about that as well, but didn't think it through. Might be the better option anyway.

Reply:
@adk3798  what do you think?